openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/remote/ServerConnector.java

```diff
@@ -171,32 +175,40 @@ private void getSecureServerConnection(String user, String pw) {
 		String[] credentials = new String[] { user, pw };
 		props.put("jmx.remote.credentials", credentials);             
 
-		// If we didn't connect keep try - the jvm being monitored might not be 
-		// ready to accept connections yet 
 		boolean connected = false;
 		int attempts = 0;
-		while (connected == false && attempts < 30) {        
+		long connectTimeout = 300L * 1000L * 1000L * 1000L;  // 300 secs in nanoseconds
+		long connectStart = System.nanoTime();
+		long connectEnd = 0L;
+		long connectElapsed = 0L;
+		while (connected == false && attempts < 30 && (connectElapsed < connectTimeout) ) {
+			Message.logOut("Attempting to connect");
 			try {    
 				connector = JMXConnectorFactory.connect(address, props);
 				connected = true;
 			} catch (IOException ie) {
-				try {           	  
-					Thread.sleep(5000);
-				} catch (InterruptedException e) {
-					Message.logOut("The sleeping profiler was interrupted");
-					e.printStackTrace();
-				}
-
-				Message.logOut("Monitored VM not ready at " + DateFormat.getDateTimeInstance().format(new Date()) +" (attempt " + attempts + ").");
-				Message.logOut("Wait 5 secs and trying again...");	
+				// If we didn't connect keep trying - the jvm being monitored might not be 
+				// ready to accept connections yet 
+				connectEnd = System.nanoTime();
+				connectElapsed = connectEnd - connectStart;
 
 				attempts++;           	
 				connected = false;
 
-				if (attempts == 30) {
-					Message.logOut("Failed to connect to Monitored VM after 30 attempts - giving up.  Connection Exception received is below:");	 
+				if (connectElapsed > connectTimeout || attempts == 30 ) {
+					Message.logOut("Failed to connect to Monitored VM after " + attempts + " attempts in " + (connectElapsed / 1000000000) + " seconds - giving up.  Connection Exception received is below:");	 
```

Comment:
magic number could be extracted

Reply:
Which 'magic number' are you referring to, please?

Reply:
The 30 and the 1000000000 could both be extracted?

Reply:
Thanks for clarifying.  By 'extracted' you mean something like
```
maxAttempts = 30;
nanosecondsPerSecond = 1000L * 1000L * 1000L;
```
?

Reply:
Something like that yes, it's probably just personal programming style preference, but makes it easier to parameterize/override later.